src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/CommonRules/BlockSizeRuleTest.cs

```diff
@@ -21,7 +21,7 @@ public BlockSizeRuleTest()
         [Fact]
         public async Task RunAsync_BadBlockWeight_ThrowsBadBlockWeightConsensusErrorExceptionAsync()
         {
-            this.ruleContext.ValidationContext.Block = GenerateBlockWithWeight((this.options.MaxBlockWeight / this.options.WitnessScaleFactor) + 1, TransactionOptions.All);
+            this.ruleContext.ValidationContext.Block = GenerateBlockWithWeight( (int) (this.options.MaxBlockWeight / this.options.WitnessScaleFactor) + 1, TransactionOptions.All);
```

Comment:
Casts everywhere are painful: I tried to make everything consistent with BlockDefinition. 

Opened issue to fix this here: https://github.com/stratisproject/StratisBitcoinFullNode/issues/1740